src/gui/render/array_buffer/array_buffer.hpp

```diff
@@ -174,34 +171,52 @@ class ArrayBuffer {
     GLuint divisor_;   // For instancing usually 0, 1
 
  public:
- /**
-  * @brief Default constructor
- */
+    /**
+     * @brief Default constructor
+     */
     inline ArrayBuffer() : divisor_(0) {}
 
     /**
      * @brief Construct ArrayBuffer with data
-     * 
+     *
      * @param std::vector<T>& data data to send to GPU
-    */
-    inline ArrayBuffer(const std::vector<T>& data) : ArrayBuffer(data, 0) {}
+     */
+    inline explicit ArrayBuffer(const std::vector<T>& data) : ArrayBuffer(data, 0) {}
 
     /**
      * @brief Construct ArrayBuffer with data and divisor
-     * 
+     *
      * @param std::vector<T>& data data to send to GPU
      * @param GLuint divisor go look up instancing
-    */
-    inline ArrayBuffer(const std::vector<T>& data, GLuint divisor) {
+     */
+    inline explicit ArrayBuffer(const std::vector<T>& data, GLuint divisor) {
         update(data, divisor);
     };
 
+    /**
+     * @brief Construct ArrayBuffer with data and divisor
+     *
+     * @param std::vector<T>& data data to send to GPU
+     * @param GLuint divisor go look up instancing
+     */
+    inline ArrayBuffer(std::initializer_list<T> data, GLuint divisor) :
+        divisor_(divisor) {
+        update_data(data.begin(), data.size());
```

Comment:
```suggestion
        update_data(data.data(), data.size());
```

Reply:
the `std::initializer_list<T>` has no `data` method. I think it exists in some header, but I don't want to incude it.

Reply:
> I think it exists in some header, but I don't want to incude it.

Don't be a lazy programmer.

It doesn't have a `data()` method though. You are right.